mixed/flink/flink-common/src/test/java/com/netease/arctic/flink/catalog/TestLegacyMixedCatalog.java

```diff
@@ -572,4 +572,84 @@ private void insertValue() {
 
     sql("DROP TABLE default_catalog.default_database." + TABLE);
   }
+
+  @Test
+  public void testAlterUnKeyTable() throws Exception {
```

Comment:
Why this is a legacy test?

Reply:
#2344 Rename By @xieyi888 Maybe merge to TestMixedCatalogTablePartitions or future test?

Reply:
ye, I see
No problem

Reply:
It seems a little confusion
As I know, LEGACY Type is for this case
```
CREATE CATALOG <catalog_name> WITH (
  'type'='arctic',
); 
```

we will have four catalog type in the future
https://github.com/NetEase/amoro/blob/d0f613fc26aafded9f40ed489a8df3dce4dfb86a/mixed/flink/flink-common/src/main/java/com/netease/arctic/flink/catalog/factories/CatalogFactoryOptions.java#L34
![image](https://github.com/NetEase/amoro/assets/105206850/934de192-ea97-43c1-8252-d36ec03e71b4)

So we have Three Test case:

- TestLegacyMixedCatalog (tset case for type='arctic')

- TestMixedCatalog containe (tset case for type='mixed_iceberg' , type='mixed_hive')

- FlinkUnifiedCatalogITCase (tset case for type='unified')

But now, some test cases for type='mixed_iceberg' type='mixed_hive' is below TestLegacyMixedCatalog. It make confusion.
https://github.com/NetEase/amoro/blob/d0f613fc26aafded9f40ed489a8df3dce4dfb86a/mixed/flink/flink-common/src/test/java/com/netease/arctic/flink/catalog/TestLegacyMixedCatalog.java#L111
![image](https://github.com/NetEase/amoro/assets/105206850/68e90bef-49af-4c3b-8921-617fe6abafaf)

Maybe we could do some adjust in the future and make TestCase for different Catalog type more clearly @YesOrNo828 HDYT